### PR TITLE
Address issues from v2.8.0 testing, misc fixes/changes

### LINKED ISSUE #1975: autograde_done should create user directory if necessary
Currently, the user directory is only created (via `create_user_directory_and_return_handin_file_path`) within the method `save_file`.

However, if a user manually calls `sendJob`, the directory will not be created. This causes an error when `autograde_done -> autogradeDone` attempts to save feedback to `autograde_feedback_path`.

Thus, `autograde_done` should call `create_user_handin_directory`

### PR DESCRIPTION
**Pull request by damianhxy**
### Summary generated by Reviewpad on 04 Jan 24 17:54 UTC
This pull request includes several changes across multiple files:

1. Added new error messages, help messages, fields, and sections to enhance the functionality and user experience of the Doorkeeper application.
2. Added a new method to create a user hand-in directory and return the path to the autograde feedback file in the `submission.rb` file.
3. Updated the rake task command syntax for promoting a user to the admin level in the `no_user.html.erb` file.
4. Modified the HTML template for the "no_user" page by updating the rake task command syntax for promoting a user to the admin level on line 67.
5. Updated the javascript code and structure of collapsible sections in the `manage.html.erb` file.
6. Modified the request header for a specific endpoint in the `tango-rest.md` file.
7. Added a new file, `collapsible.js`, to initialize and configure a collapsible menu feature.
8. Improved the logic for redirecting and handling unreleased assessments based on the user's role in the `index.html.erb` file.
9. Modified the `autogradeDone` method and related variables in the `assessment_autograde_core.rb` file.

These changes introduce new functionality, improve existing features, and enhance the user experience in the codebase.

## Description

Address issues uncovered during [v2.8.0 testing](https://docs.google.com/spreadsheets/d/1K89NjIgr_98V6Z0JFV6d4BvoUD8dz5RczY4jfZokrdc/edit#gid=0)
* Correct rake command described for promotion to admin
* "unreleased" badge no longer clickable for CAs (since they can't edit an assessment)

Other changes
* Fix Tango documentation for `info` endpoint
* Create user handin directory on `autograde_done` if necessary
* Change attribute aria-role -> role (aria-role isn't an actual attribute)
* Open first item of first accordion on view assessment / manage course page (done programmatically to avoid hardcoding the HTML)
* Remove admin options from manage course page
* Fix error handling for downloadAll
* Fix redirect logic within `set_assessment`: `action: :index` was ambiguous, so if viewing a submission route of an invalid assessment, it redirects to submission#index infinitely

Maintenance
* Extract common aria code from `assessments/show.html.erb` and `courses/manage.html.erb` pages
* Remove empty classes attributes and redundant `then`s
* Use path helpers where applicable
* Update doorkeeper translations (by running `rails g doorkeeper:install`)
* Remove unused variables
* Remove redundant gitignores

## Motivation and Context

Resolves some outstanding issues.

Closes #1975 and https://github.com/autolab/Tango/issues/136

## How Has This Been Tested?

- Verify that `rake admin:promote_user[{email addr}]` is the correct command to promote a user to admin
- Unreleased assessment badge no longer clickable for CAs
- Links on assessment view page still work / same as before
- First item of first accordion opens on assessment landing page / course manage page (check as instructor, CA, and student). Also the accordions work in general.
- Autograding still works

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://docs.autolabproject.com/)
- [ ] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -223,6 +223,13 @@ def autograde_feedback_path
     old_autograde_feedback_path
   end
 
+  def create_user_directory_and_return_autograde_feedback_path
+    create_user_handin_directory
+
+    File.join(assessment.handin_directory_path, course_user_datum.email,
```
Comment (20wildmanj) [inline app/models/submission.rb:229]: nit: maybe consolidate autograde feedback path into a function or something? (duplicated on 214)

File: app/helpers/assessment_autograde_core.rb [hunk 2]
```diff
@@ -357,7 +357,7 @@ def autogradeDone(submissions, feedback)
     ass_dir = @assessment.folder_path
```
Comment (20wildmanj) [inline app/helpers/assessment_autograde_core.rb:357]: nit: not introduced by pr but seems like this variable is unused?

**Comment by 20wildmanj** [comment]
NGL I don't really like the change to coalesce the accordions, I feel like having at least a little bit of white space looks better / easier to see the delineation than smushing them together.


before:
Screenshot 2024-01-05 at 11 19 28 PM

after:
Screenshot 2024-01-05 at 10 56 53 PM

Screenshot 2024-01-05 at 10 57 13 PM

File: app/models/submission.rb (hunk 1 above)
Comment (damianhxy) [reply app/models/submission.rb:229]: I was following what `create_user_directory_and_return_handin_file_path` did, which also duplicates the logic

I'll extract the new paths into helper methods

File: app/helpers/assessment_autograde_core.rb (hunk 2 above)
Comment (damianhxy) [reply app/helpers/assessment_autograde_core.rb:357]: Removed

**Pull request merged**